practical/matlab/idr0021_steps.m

```diff
@@ -0,0 +1,203 @@
+% of each exercise, right-click and "Evaluate Selection". 
+% Then proceed to the next exercise.
+% The exercises build on top of each other (later exercises
+% cannot be run unless the previous exercises have been 
+% executed successfully). If you get stuck, clear the workspace
+% and try again from the beginning.
+
+% Exercise 1
+% Connect to OMERO and print your group ID.
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+client = loadOmero('outreach.openmicroscopy.org');
+session = client.createSession('USER', 'PASSWORD');
+client.enableKeepAlive(60);
+eventContext = session.getAdminService().getEventContext();
+groupId = eventContext.groupId;
+disp(groupId);
+
+
+% Exercise 2
+% List the images of a particular dataset.
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+datasetId = MATLAB_DATASET_ID;
+loadedDatasets = getDatasets(session, datasetId, true);
+dataset = loadedDatasets(1);
+datasetName = dataset.getName().getValue();
+disp(datasetName)
+datasetImages = getImages(session, 'dataset', datasetId);
+for i = 1 : length(datasetImages)
+    image = datasetImages(i);
+    fprintf('%s , %i\n', image.getName().getValue(), image.getId().getValue());
+end
+
+
+% Exercise 3
+% Get the target protein name (== name of the relevant channel).
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+image = datasetImages(1); % Pick one image from the dataset (they all have the same target protein)
+annotations = getObjectAnnotations(session, 'map', 'image', image.getId().getValue());
+% Iterate through all map annotations ('key-value pairs')
+for j = 1 : length(annotations)
+    rows = annotations(j).getMapValue();
+    for k = 0 : rows.size() - 1
+        if rows.get(k).name == 'Antibody Target'
+            target = rows.get(k).value;
+            break;
+        end
+    end
+end
+fprintf('Target protein: %s\n', target);
+
+
+% Exercise 4
+% Determine the channel indices of the relevant channel.
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+for i = 1 : length(datasetImages)
+    image = datasetImages(i);
+    channels = loadChannels(session, image);
+    for j = 1 : numel(channels) 
+        channel = channels(j);
+        channelId = channel.getId().getValue();
+        lc = channel.getLogicalChannel();
+        channelName = lc.getName().getValue();
+        if channelName == target
+            channelIndex = j;
+            disp(channelIndex);
+        end
+    end
+end
+
+
+% Exercise 5
+% Perform image segmentation on one image.
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+fprintf('Using image: %s, channel: %s (index: %i)\n', image.getName().getValue(), target, channelIndex)
+z = 0;
+t = 0;
+plane = getPlane(session, image, z, channelIndex - 1, t);
+threshNstd = 6;
+minPixelsPerCentriole = 20;   % minimum size of objects of interest
+vals = reshape(plane, [numel(plane), 1]);   % reshape to 1 column
+mean1 = mean(vals);
+std1 = std(vals);
+% images are mostly background, so estimate threshold using basic stats
```

Comment:
``The centrioles in each image occupy only relatively small areas, the rest is background.``

For the rest of the sentence, use some authoritative explanation from literature about the formula 
``thresh1 = mean1 + threshNstd * std1``

Reply:
the 6xStdev is simply getting rid of all the bckgr pixels - according to the rule https://en.wikipedia.org/wiki/68%E2%80%9395%E2%80%9399.7_rule, only 0.3 percent of bckgr pixels are catched together with the centrioles. This assumes that the centrioles are always much brighter than that. This is not always the case, but at least we are sure never to catch any background.